### ISSUE TITLE
Missing sort function in maps

### ISSUE THREAD

**Issue by ricma** [chiquitinxx/grooscript#53]
Hi there,

very nice converter! Thanks for all your effort on this, it's really great.

Using the [online converter](http://grooscript.org/conversions.html) with the following code

``` groovy
assert [(1): "A", (2): "B"].sort { 
    e1, e2 -> e2.getKey() <=> e1.getKey() }.values() as List == ["B", "A"]
```

but ran into the error message:

```
Fail evaluating Js Script! - sun.org.mozilla.javascript.EcmaError: ReferenceError: 
  "sort" is not defined. (#1(eval)#1) in #1(eval) at line number 1
```

Is this a missing feature? it seems similar to issue #43 where the count function on Lists was missing.

Best regards,
M.

**Comment by chiquitinxx** [comment]
Yes, not implemented. Do you want to implement it?

**Comment by ricma** [comment]
Hey there,

I already forked the repo -- Let me see if I can come up with a solution.

BR,
M.

**Comment by ricma** [comment]
After working myself into grooscript a bit I found the spot to implement it. However, as Javascript itself lacks a LinkedHashMap equivalent, there won't be a simple solution here.

Things which spring to my mind are

1. Implementing a custom map which keeps track of the internal ordering by a separate array. However, then the usage might be a bit odd an I would not want to impose this implementation on anyone using plain objects in the Javascript result.

2. One could use the above specialization only as a return type when calling sort() on a map. But this would also break the expectation from a Groovy point of view where all Maps are LinkedHashMaps.

3. We all just wait for [ECMAScript 6](http://stackoverflow.com/a/14645486) to come up with a solution we can implement against.

**Comment by chiquitinxx** [comment]
There is already an implementation in grooscript.js, you have to add the
sort method to it. Can take a look at other implemented method, also how
tests works. Ask any questions, Im happy to help

**Comment by ricma** [comment]
Hi there,

Thanks for the hint. Yes, I know. I found the implementation for sort on lists and tried to mimic the same approach for maps. I also found the tests and [wrote my own](https://github.com/chiquitinxx/grooscript/compare/master...ricma:add-feature-sort-to-maps) ... but it's still work in progress, the closure calling does not seem to work the way I want it to. That is why there is so much debug output -- don't consider it a Pull Request yet ;-).

But there is a fundamental difference between Arrays and Maps in Javascript: While the sort function for Lists relies on the fact  that it can sort the underlying Javascript Array there is no such equivalence for Groovy Maps and Javascript Objects. I can write a sort function for Maps, I can apply the sort function on the items -- but there is no genuine way of keeping the Javascript Object, which backs the Grooscript Map implementation underneath, sorted.

But I might have overseen something there.

BR,
M.

PS: As a side note: I will be rather busy the next two to three weeks -- so don't be surprised if I won't be able to answer immediately. I don't mean to be rude, it will just be due to a lack of time.
